core/utils/poly-look/src/visualisations/charts/bubble-charts/bubbleCluster/bubbleCluster.js

```diff
@@ -165,9 +174,124 @@ export class BubbleCluster extends Chart {
       .style("font-family", "Jost Medium")
       .style("font-weight", "500")
       .attr("fill", this._textColor)
-      .attr("transform", (d) =>
-        d.data.icon ? `translate(${d.x},${d.y})` : ""
-      );
+      .attr("transform", (d) => (d.data.icon ? `translate(${d.x},${d.y})` : ""))
+      .on("click", this._onBubbleClick);
+  }
+
+  _addLabel(bubbleGroups) {
+    const label = this._label;
+    const color = this._textColor;
+    const chart = this.chart;
+    bubbleGroups.each(function (node) {
+      if (label(node)) {
+        const labelGroup = d3
+          .select(this)
+          .append("g")
+          .attr("class", "label-group");
+        const text = labelGroup
+          .append("text")
+          .attr("x", 0)
+          .attr("y", (d) => -d.r - labelMargin)
+          .attr("text-anchor", "middle")
+          .attr("fill", this._textColor)
+          .text(label);
+
+        text.each((_, i, nodes) => {
+          const bBox = nodes[i].getBBox();
+          labelGroup
+            .append("rect")
+            .attr("x", bBox.x - nodeLabelBoxPaddingX)
+            .attr("y", bBox.y - nodeLabelBoxPaddingY)
+            .attr("width", bBox.width + nodeLabelBoxPaddingX * 2)
+            .attr("height", bBox.height + nodeLabelBoxPaddingY * 2)
+            .attr("fill", defaultNodeLabelBoxColor)
+            .attr("opacity", defaultNodeLabelBoxOpacity)
+            .attr("rx", nodeLabelBoxBorderRadius);
+        });
+
+        const rect = labelGroup.select("rect");
+
+        const line = labelGroup
+          .append("line")
+          .attr("x1", 0)
+          .attr("x2", 0)
+          .attr("y1", (d) => -d.r - 2)
```

Comment:
2 and 7, and also 10, are repeated several times. Again, is it possible to convert that into a named constant?
Also, if possible, consider a function generator that creates this kind of functions according to some external value. For instance, the function at 218 is (-1*)function at 237. Clarifying why that happens would make easier to understand how these functions are built.